src/config/settings.js

```diff
@@ -62,6 +65,61 @@ function init() {
   return settings;
 }
 
+function safeFileTest(path, conditions) {
+  try {
+    fs.accessSync(path, conditions);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+function fileExists(path) {
+  try {
+    fs.statSync(path);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+function fileIsWritable(path) {
+  try {
+    fs.accessSync(path, fs.constants.W_OK);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+function qq(fullpath) {
+  return /\s/.test(fullpath) ? `"${fullpath}"` : fullpath;
+}
+
+function parseSaveError(err) {
+  let msg = err.toString();
+  console.log(`settings save error: ${msg}`);
+  let p = new RegExp(`^Error:\\s*${err.code}:\\s*(.*?),\\s*${err.syscall}\\s+'?${err.path}`);
```

Comment:
Not really happy with string parsing, but I couldn't find a better alternative. So I guess that's a :+1:.

Reply:
The node way is to write and publish a module to do this in a cross-platform way, right? Seems like overkill until we go multi-platform